main.cpp

```diff
@@ -91,6 +91,116 @@ good to go!
  Wait for my code review.
  */
 
+ #include <iostream>
+
+struct FloatType
+{
+    float add(float lhs, float rhs);
+    float subtract(float lhs, float rhs);
+    float multiply(float lhs, float rhs);
+    float divide(float lhs, float rhs);
+};
+
+float FloatType::add(float lhs, float rhs)
+{
+    return lhs + rhs;
+}
+
+float FloatType::subtract(float lhs, float rhs)
+{
+    return lhs - rhs;
+}
+
+float FloatType::multiply(float lhs, float rhs)
+{
+    return lhs * rhs;
+}
+
+float FloatType::divide(float lhs, float rhs)
+{
+    if (lhs == 0.0f || rhs == 0.0f)
+    {
+        std::cout << "\nwarning, floating point division by zero returns 'inf' !" << std::endl;
+    }
+    return lhs / rhs;
+}
+
+//=================================================
+
+struct DoubleType
+{
+    double add(double lhs, double rhs);
+    double subtract(double lhs, double rhs);
+    double multiply(double lhs, double rhs);
+    double divide(double lhs, double rhs);
+};
+
+double DoubleType::add(double lhs, double rhs)
+{
+    return lhs + rhs;
+}
+
+double DoubleType::subtract(double lhs, double rhs)
+{
+    return lhs - rhs;
+}
+
+double DoubleType::multiply(double lhs, double rhs)
+{
+    return lhs * rhs;
+}
+
+double DoubleType::divide(double lhs, double rhs)
+{
+    if (lhs == 0.0 || rhs == 0.0)
+    {
+        std::cout << "\nwarning, floating point division by zero returns 'inf' !" << std::endl;
+    }
+    return lhs / rhs;
+}
+
+//===============================================
+
+struct IntType
+{
+    int add(int lhs, int rhs);
+    int subtract(int lhs, int rhs);
+    int multiply(int lhs, int rhs);
+    int divide(int lhs, int rhs);
+};
+
+int IntType::add(int lhs, int rhs)
+{
+    return lhs + rhs;
+}
+
+int IntType::subtract(int lhs, int rhs)
+{
+    return lhs - rhs;
+}
+
+int IntType::multiply(int lhs, int rhs)
+{
+    return lhs * rhs;
+}
+
+int IntType::divide(int lhs, int rhs)
+{
+    if (lhs == 0 || rhs == 0)
```

Comment:
it doesn't matter if `lhs` is 0.  it only matters if `rhs` is 0.  if `rhs` is 0, then you'll actually have a division by zero.
- [ ] revise